salt/executors/venv.py

```diff
@@ -0,0 +1,13 @@
+"""
+Wipe PYTHONHOME environment variable for venvjailed salt-minion
+"""
+import os
+
+
+def execute(opts, data, func, args, kwargs):
+    """
+    Wipe PYTHONHOME environment variable for venvjailed salt-minion
+    """
+    if os.environ.get("VIRTUAL_ENV", None) == os.environ.get("PYTHONHOME", ""):
```

Comment:
I am confused with both have different defaults (one `None` and the other `""`). I would be super super simple here:

* Do not upstream the executor in salt, is a very specific use case and fairly useless outside the venv-salt-minion
* Add the executor only inside the venv-salt-minion package, and add the configuration snipped that activate it inside the venv
* Make this executor a two line code:

```python
# No exception if it is not there
os.environ.pop("PYTHONHOME", None)
return __executors__["direct_call.execute"](opts, data, func, args, kwargs)
```
* Drop the tests

Reply:
@aplanas, yes you are right it's useless for any case other than `venv-salt-minion`, but I suppose it's better to have it here (not in the upstream salt for sure) just to make it easier to maintain.
`os.environ.get("VIRTUAL_ENV", None) == os.environ.get("PYTHONHOME", "")` was an attempt to prevent the conditional step if both are not present. It's safe for sure, but here I'm trying to prevent wiping `PYTHONHOME` if the executor is used outside the `venv-salt-minion`. I can't imagine such case anyway and probably we don't really need this check.

Reply:
> but I suppose it's better to have it here (not in the upstream salt for sure) just to make it easier to maintain

Uhm I would not put in the salt package, as the code would be distributed for all the users and will require an update of the salt package each time that we want to fix something in the venv-salt-minion package.

Also making the two-line version a thing inside the venv-salt-minion open the door to provide more similar fixes for this specific use case without any change. For example if we want to adjust the "pip" command we can influence from here.

> `os.environ.get("VIRTUAL_ENV", None) == os.environ.get("PYTHONHOME", "")` was an attempt to prevent the conditional step if both are not present.

Sure. I was wondering about the different defaults (like making both `None`, or both `""`). And this is not needed if lives in the venv (as we want to be removed always)

Reply:
Agree, closing the PR without merging